### PR TITLE
chore (node-fetch): remove redundant types

### PR DESCRIPTION
**Pull request by 43081j**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).
- [x] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [x] Delete the package's directory.
- [x] Add it to `notNeededPackages.json`.

---

types shipped since 3.0.0:
https://unpkg.com/browse/node-fetch@3.0.0/@types/index.d.ts

the source packages of the dependents here in DT still use node-fetch 2.x, so i've set them to depend on `@types/node-fetch` rather than bumping them up to the shipped 3.x types.

they're probably the same but until those packages upgrade to 3.x, it seems sensible to keep using the types package.

also, there's a bunch that pull things like `Request` in from node-fetch. However, looking at the source of those packages, they actually use a package called `minipass-fetch` instead (nothing to do with node-fetch). Those types are probably all wrong and should pull from `minipass-fetch` instead, though that has no types itself.

### CODE REVIEW

**Comment by 43081j** [comment]
@andrewbranch any idea what to do in this situation? the tests of `proto-fetch` depend on node-fetch, but the types themselves don't.

im going to remove it from the test since it isn't actually testing much (if anyone disagrees, let me know, open to suggestions).

but if we didn't have that as an option, what would we do? for future reference

**Comment by tpluscode** [comment]
Hey @43081j, please not that we've already been down this path

PR #55533 attempted to remove `node-fetch` but the [discussion](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/55533#issuecomment-1034075119) led to restoring them. #58693 followed, to bring these types back

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## frisby ([<kbd>unpkg</kbd>](https://unpkg.com/browse/frisby@latest/))
was missing the following properties:
1. delete
1. delete
## make-fetch-happen ([<kbd>unpkg</kbd>](https://unpkg.com/browse/make-fetch-happen@latest/))
was missing the following properties:
1. FetchError
1. Headers
1. Request
1. Response
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 25fdeb47e6cbdec859d8981c7154df46fae3d458
</p>

**Comment by andrewbranch** [comment]
> the tests of `proto-fetch` depend on node-fetch, but the types themselves don't

I always request changes when I see this in a PR. It’s pretty hard to enforce new rules in DT, but I would remove test-only dependencies everywhere possible.

**Comment by glasser** [comment]
Agreed with @tpluscode. `node-fetch` is an incredibly popular npm package, and we're not yet at a state where everybody's build environment supports ESM only, which is required by v3.  Many many many people will still be using v2 for a while.  Looking at the last 7 days stats on npmjs.com, v2.6.7 has 24 times more downloads than v3.2.10. Heck, v2.6.6 (which was only the latest version for a few months) has more downloads than v3.2.10.

I'm a maintainer for `node-fetch` but I'm not enough of a DT expert to know what to actually do here; eg, I cannot personally close the PR.

**Review by glasser — CHANGES_REQUESTED** [review]
As described in my previous comment, removing the type definitions for a wildly popular older major version is not a good idea. I'm not really asking for changes, just to have this closed, though I don't know how to do that.

**Comment by glasser** [comment]
@43081j though to be clear — I appreciate the effort you've put into this. Where do you think we could put a clear explanation of why we continue to maintain these files, so that others won't try to delete them?

**Comment by jakebailey** [comment]
I guess my main question is whether or not these types actually need to be updated in the long term; if they aren't, then deprecating the package probably won't hurt things, because either users install node-fetch v3 and it works, or they install node-fetch v2 and there's an error message that says to install `@types/node-fetch`.

Maybe it's not so nice that installing the package will say that it's not needed at first install, I suppose.

**Comment by glasser** [comment]
I've personally found that the state of fetch typing across TS (these node-fetch typings, the DOM typings that are bundled, etc) has had more subtle inconsistencies and needed incremental improvements than you might expect. Also worth noting that 5 patch releases of v2 have been published since v3 was released, though I don't know if any affected the API shape (types).

**Comment by glasser** [comment]
(#60808 is a concrete example of the sort of incremental change that was needed recently due to the subtleties of the different implementations of this API.)

**Comment by andrewbranch** [comment]
There’s not much harm in keeping them around if folks are concerned. I’m guessing @43081j was just doing some housekeeping (much appreciated) and isn’t attached to deleting them.

**Comment by 43081j** [comment]
I have no strong opinion one way or the other about keeping these around. 

However I do feel like there's a lot of misunderstanding going on about how DT works. 

Let's say that nobody ever moves to 3.x, they can always stay on the older, already published version of the types quite easily. 

The only possible situation we have where that falls over is if the currently published types are incorrect, as we can't go back and update them once they're gone. 

The last attempt at removing them failed specifically because of this. The source package didn't change, the types were just incomplete all along. 

I did manually read through the source package and the types to make sure we haven't missed anything else before removing them, and didn't find anything obvious. But of course it's easy to miss small things. 

Tldr: if the types are complete, we should remove them and everything will be fine. But if there's any chance they're still wrong and will need updating, we shouldn't.

**Comment by glasser** [comment]
Agreed. I'm just saying that there has been a history so far of this relatively large API that's trying to match a complex W3C API needing minor typing adjustments. (And as aspects of it get implemented in Node core I suspect more changes to keep those more compatible with each other may be nice too.)

**Comment by 43081j** [comment]
@glasser 2.x won't be updated anymore though, right? at the source

`agent` was added explicitly to node-fetch and simply missed out of the types by accident.

assuming 2.x never changes again, it would be sensible to keep these types around only if we suspect they may still be wrong i think. if there's any chance we will have to update them again, there's no point removing them yet.

i did a quick pass but can't guarantee we haven't missed anything.
